sandbox/modules/identity/src/main/java/org/opensearch/identity/rest/user/resetpassword/ResetPasswordRequest.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.identity.rest.user.resetpassword;
+
+import org.opensearch.action.ActionRequest;
+import org.opensearch.action.ActionRequestValidationException;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.xcontent.ToXContentObject;
+import org.opensearch.common.xcontent.XContentBuilder;
+
+import java.io.IOException;
+
+import static org.opensearch.action.ValidateActions.addValidationError;
+
+/**
+ * Request to reset an internal user password
+ */
+public class ResetPasswordRequest extends ActionRequest implements ToXContentObject {
+
+    private String username;
+    private String oldPassword;
+    private String newPassword;
+    private String newPasswordValidation;
```

Comment:
In UX passwords are asked for confirmation, but this is because they potentially being typed by hand.  This is an API that will called by code, doesn't seem like its of use.  Note; this doesn't prevent a page in OpenSearch Dashboards or another UI from requiring the user type the password twice, but it shouldn't be part of the API requirements.

Take a look at this documentation for a REST password API https://learn.microsoft.com/en-us/graph/api/authenticationmethod-resetpassword?view=graph-rest-beta&tabs=http

I would notable callout that it doesn't require a new password, as it will autogenerate a password if one wasn't provided.

Reply:
Good catch! I have actually removed this field, seems like I need to remove this line too